packages/core/src/services/documents/saveLinkedDataset.ts

```diff
@@ -0,0 +1,43 @@
+import { eq } from 'drizzle-orm'
+
+import { Dataset, DocumentVersion, LinkedDataset } from '../../browser'
+import { database } from '../../client'
+import { Result, Transaction, TypedResult } from '../../lib'
+import { documentVersions } from '../../schema'
+
+export async function saveLinkedDataset(
+  {
+    document,
+    dataset,
+    data,
+  }: {
+    document: DocumentVersion
+    dataset: Dataset
+    data: {
+      rowIndex: number
+      inputs: LinkedDataset['inputs']
+      mappedInputs: LinkedDataset['mappedInputs']
+    }
+  },
+  trx = database,
+): Promise<TypedResult<DocumentVersion, Error>> {
+  const prevLinked = document.linkedDataset ?? {}
+  return await Transaction.call(async (tx) => {
+    const linkedDataset = {
+      ...prevLinked,
+      [dataset.id]: {
+        rowIndex: data.rowIndex,
+        inputs: data.inputs,
+        mappedInputs: data.mappedInputs,
+      },
+    }
```

Comment:
Do a test for this. We have to ensure old dataset mappings are preserved and only the dataset ID passed is overwritten